asserts/fetcher.go

```diff
@@ -106,13 +106,25 @@ func (f *fetcher) Fetch(ref *Ref) error {
 	return f.chase(ref, nil)
 }
 
-// fetchAccountKey behaves like Fetch for the account-key with the given key id.
-func (f *fetcher) fetchAccountKey(keyID string) error {
+// fetchAccountKey behaves like Fetch for the account-key with the given key id
+// and related authority-delegation if needed for the provided
+// authority account-ids.
+func (f *fetcher) fetchAccountKey(keyID, signatoryID, authorityID string) error {
 	keyRef := &Ref{
 		Type:       AccountKeyType,
 		PrimaryKey: []string{keyID},
 	}
-	return f.Fetch(keyRef)
+	if err := f.Fetch(keyRef); err != nil {
+		return err
+	}
+	if signatoryID == authorityID {
+		return nil
+	}
+	delegationRef := &Ref{
```

Comment:
Nitpick, maybe add a clarification comment: `Signatory ID is never empty (it is equal to authorityID if not set)`.